android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirTaskPlanWorker.kt

```diff
@@ -58,14 +58,6 @@ constructor(
 
   override suspend fun doWork(): Result {
     return withContext(dispatcherProvider.io()) {
-      val appConfig =
-        configurationRegistry.retrieveConfiguration<ApplicationConfiguration>(
-          ConfigType.Application
-        )
-      val batchSize = appConfig.taskBackgroundWorkerBatchSize
-      val lastOffset =
-        sharedPreferencesHelper.read(key = WORK_ID.lastOffset(), defaultValue = "0")!!.toInt()
-
```

Comment:
The logic for batching was introduced in this [PR](https://github.com/opensrp/fhircore/pull/2351) as part of the performance enhancements for background workers.
Do you mind explaining why it has been removed?

Reply:
Can you respond to @Rkareko question above? Will the refactored filter improve the query performance? I think regardless of the refactor done on the filters, the query will still run against all the available Tasks in the app.

Reply:
The result of the query for Tasks is not predictable and not always same amount of tasks and hence we can not use the offset/batch size broken down across different executions. There are other Jobs which mark tasks expired as well, or some times tasks can also be completed by form submissions, and we can in future have a functionality that moves task from one state to another. A task itself can have or be eligible for a different state at different point of time. Lets say today we have 1000 tasks which resulted with given params in query with batch size of 50. We would update 50 tasks and next might be updated after 1 hour, or maybe 2 hours when this is executed. But in the mean time 100 Tasks were expired by other service, 100 were completed by form submission or whatso ever. Now when after an hour query runs the tasks would be 800 and offset starts from 50. We have no way to tell what Tasks we have missed due to sorting order, and status updates. Also as opposed to getting ALL tasks of given statuses, we are now filter tasks with status and due date which would be lesser in  number, hence making lesser impact on performance. However, if still we want to improve performance the Task query count and objects should be predictable, and with same entities and in same order

Reply:
Resolved this conversation in a separate meeting. We confirmed @maimoonak's observation above.